Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/organiser/ui/TestTraceLocationsFragment.kt

```diff
@@ -0,0 +1,104 @@
+package de.rki.coronawarnapp.test.organiser.ui
+
+import android.os.Bundle
+import android.view.View
+import android.view.accessibility.AccessibilityEvent
+import android.widget.Toast
+import androidx.appcompat.widget.Toolbar
+import androidx.core.view.isGone
+import androidx.fragment.app.Fragment
+import com.google.android.material.transition.Hold
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.TraceLocationOrganizerTraceLocationsListFragmentBinding
+import de.rki.coronawarnapp.test.menu.ui.TestMenuItem
+import de.rki.coronawarnapp.util.DialogHelper
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.lists.decorations.TopBottomPaddingDecorator
+import de.rki.coronawarnapp.util.lists.diffutil.update
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
+
+class TestTraceLocationsFragment : Fragment(R.layout.trace_location_organizer_trace_locations_list_fragment), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: TestTraceLocationsViewModel by cwaViewModels { viewModelFactory }
+    private val binding: TraceLocationOrganizerTraceLocationsListFragmentBinding by viewBindingLazy()
+    private val traceLocationsAdapter = TestTraceLocationsAdapter()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        exitTransition = Hold()
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupMenu(binding.toolbar)
+
+        binding.recyclerView.apply {
+            adapter = traceLocationsAdapter
+            addItemDecoration(TopBottomPaddingDecorator(topPadding = R.dimen.spacing_tiny))
```

Comment:
could you please give the nice shrinking animation of FAB button :) ? as in contact diary day overview screen I think
use 
```kotlin
onSrcoll{ 
extend-> if (extend) fab.extend() else fab. shrink()
}